apps/arweave/src/ar_mining_server.erl

```diff
@@ -667,6 +688,40 @@ may_be_empty_poa(not_set) ->
 may_be_empty_poa(#poa{} = PoA) ->
 	PoA.
 
+fetch_poa_from_peers(RecallByte) ->
+	Peers = ar_data_discovery:get_bucket_peers(RecallByte div ?NETWORK_DATA_BUCKET_SIZE),
+	From = self(),
+	lists:foreach(
+		fun(Peer) ->
+			spawn(
+				fun() ->
+					?LOG_INFO([{event, last_moment_proof_search},
+							{peer, ar_util:format_peer(Peer)}]),
+					case fetch_poa_from_peers(Peer, RecallByte) of
+						not_found ->
+							ok;
+						PoA ->
+							From ! {fetched_last_moment_proof, PoA}
+					end
+				end)
+		end,
+		Peers
+	),
+	receive
+		{fetched_last_moment_proof, PoA} ->
```

Comment:
```suggestion
         %% The first spawned process to fetch a PoA from a peer will trigger this `receive` and allow
         %% `fetch_poa_from_peers` to exit. All other processes that complete later will trigger the
         %% `handle_info({fetched_last_moment_proof, _}, State) ->` above (which is a no-op)
		{fetched_last_moment_proof, PoA} ->
```

Reply:
Is that comment a correct understanding of what's happening here? Are we sure the `receive` will be processed before `handle_info`? I haven't dug this deep into erlang's mailbox handling logic before

Reply:
Yep, receive will dequeue one message, the rest goes to handle_info